src/GI.h

```diff
@@ -192,10 +230,27 @@ template <class Ptcl> class GI : public Problem<Ptcl>{
                             // TODO: Modify this line for all particles that need new EoS
                             ith.setPressure(&AGranite);
                             ith.tag = 0;
-                            if(ith.id / NptclIn1Node == PS::Comm::getRank()) tar.push_back(ith);
+
+			    if (removal_list.count(index)){
+			      id += -1;
+			    }else{
+			      if(ith.id / NptclIn1Node == PS::Comm::getRank()) tar.push_back(ith);
+			    }
+			    index += 1;
                         }
                     }
                 }
+
+		std::cout << "# of mantle particles = " <<  tar.size() << std::endl;
```

Comment:
I think this is not true in MPI parallel runs. If you have multiple MPI processes, `tar` on each one of them will only have a fraction of the total number of particles. I think this should be:
```suggestion
		std::cout << "# of mantle particles = " << id << std::endl;
		const unsignd int targetMantleParticles = id;
```